Source/WebCore/dom/DocumentFragment.cpp

```diff
@@ -93,8 +93,10 @@ Ref<Node> DocumentFragment::cloneNodeInternal(Document& document, CloningOperati
 
 void DocumentFragment::parseHTML(const String& source, Element& contextElement, OptionSet<ParserContentPolicy> parserContentPolicy, CustomElementRegistry* registry)
 {
+    m_wasParsedWithFastPath = false;
```

Comment:
I'd set this after the if statement for `tryFastParsingHTMLFragment` to avoid updating it twice for when m_wasParsedWithFastPath is true.